### PR TITLE
Add metric to report protected condition status and required alerts for the same

### PR DESCRIPTION
**Pull request by ShyamsundarR**
Fixes: #711

### CODE REVIEW

**Comment by ShyamsundarR** [comment]
Needs to be tested in an e2e setup. Unsure if we have envtests for metrics at present as well.

File: controllers/drplacementcontrol_controller.go [hunk 1]
```diff
@@ -889,6 +889,28 @@ func (r *DRPlacementControlReconciler) setLastSyncBytesMetric(syncDataBytesMetri
...
+	}
+
+	log.Info(fmt.Sprintf("setting metric: (%s)", WorkloadProtectionStatus))
+
+	protected := 0
+
+	for idx := range conditions {
+		if conditions[idx].Type == rmn.ConditionProtected && conditions[idx].Status == metav1.ConditionTrue {
+			protected = 1
+		}
```
Comment (BenamarMk) [inline controllers/drplacementcontrol_controller.go:908]: (nit) add `break`

**Review by BenamarMk — APPROVED** [review]

**Comment by rakeshgm** [comment]
> Needs to be tested in an e2e setup. Unsure if we have envtests for metrics at present as well.

There are no envtests as of now.

File: controllers/drplacementcontrol_controller.go (hunk 1 above)
Comment (ShyamsundarR) [reply controllers/drplacementcontrol_controller.go:910]: Addressed, Thanks.

**Pull request merged**